2-projects/1-todo-list/script.js

```diff
@@ -1,13 +1,51 @@
 function populateTodoList(todos) {
-  let list = document.getElementById("todo-list");
-  // Write your code to create todo list elements with completed and delete buttons here, all todos should display inside the "todo-list" element.
+  let list = document.getElementById('todo-list');
+
+  // Loops through the todos array renders an element accordingly
+  todos.forEach((todo) => {
+    let todoEl = document.createElement('li');
+    todoEl.innerHTML = todo.task; // Adds a text to the li from the array of objects
+    todoEl.classList.add(
+      'list-group-item',
+      'd-flex',
+      'justify-content-between',
+      'align-items-center'
+    );
+    let spanEl = document.createElement('span');
+    spanEl.classList.add('badge', 'bg-primary', 'rounded-pill');
+    let checkEl = document.createElement('i');
+    checkEl.classList.add('fa', 'fa-check');
+    let trashEl = document.createElement('i');
+    trashEl.classList.add('fa', 'fa-trash');
+    trashEl.style.marginLeft = '5px';
+    spanEl.appendChild(checkEl);
+    spanEl.appendChild(trashEl);
+    todoEl.appendChild(spanEl);
+    list.appendChild(todoEl);
+
+    // Adds a line through conditionally on the task element text when the checked icon is clicked
+    checkEl.addEventListener('click', () => {
+      if (todoEl.style.textDecoration === '') {
```

Comment:
we don't want to check the `textDecoration` here for this if statement. This is a bad practice. What you are doing here is using the presence or absence of a style to drive the logic. You can imagine in the future if another programmer came along to change this code and thought: "instead of adding text decoration to show an item is complete, I am going to change the style to make it red instead". That's great but they now have to remember to change this logic too or else the app will break.

We would say that you have 'made your logic dependent on your style'. In web dev we almost always want to make the style dependent on the logic. Do you see what I mean?

It would be clearer as: `if (!todo.completed) { }`